### PR TITLE
fix(#494): update lib/dependencies for TS, tools; add path,lib to tsconfig

### PR DESCRIPTION
**Pull request by sbolel**
### Purpose

- Fixes https://github.com/aurelia/cli/issues/494
- Originally asked in StackOverflow: ["Aurelia bundling fails when using relative import path"][so-question]

### Summary of Changes

- Updates Typescript-related dependencies for new generated projects in [lib/dependencies.js][lib-dependencies]
- Adds `lib`, `baseUrl`, `paths` to [lib/resources/content/tsconfig.json][content-tsconfig]
- Updates `aurelia-tools` to `^1.0.0` in both [package.json][package] and [lib/dependencies.js][lib-dependencies][so-question]

[lib-dependencies]: https://github.com/aurelia/cli/compare/master...sbolel:fix/494/bundle-non-relative-path?expand=1#diff-b873c4a67f750b273496ccc075da8b06
[content-tsconfig]: https://github.com/aurelia/cli/compare/master...sbolel:fix/494/bundle-non-relative-path?expand=1#diff-c7f84dc92dc0fb2ffa7fde3bddcd2b39
[package]: https://github.com/aurelia/cli/compare/master...sbolel:fix/494/bundle-non-relative-path?expand=1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2
[so-question]: http://stackoverflow.com/questions/42411567/aurelia-bundling-fails-when-using-relative-import-path

### TODOs

- [ ] Fix `gulp-typescript: ts(tsProject) has been deprecated`. 
  - Bug: https://github.com/aurelia/cli/issues/494#issuecomment-282222424
  - Fix: https://github.com/aurelia/cli/issues/468#issuecomment-274923144
- [ ] Fix type errors
  - Bugs: https://github.com/aurelia/cli/issues/494#issuecomment-282229089, https://github.com/aurelia/cli/issues/494#issuecomment-282232932
  - Fix: https://github.com/aurelia/cli/issues/494#issuecomment-282244119
- [x] remove typings -- #457.

### CODE REVIEW

**Comment by sbolel** [comment]
For reference, here's the impact of these changes on `au build`:

Before:

```bash
$ au build
Starting 'readProjectConfiguration'...
Finished 'readProjectConfiguration'
Starting 'processMarkup'...
Starting 'processCSS'...
Starting 'configureEnvironment'...
Finished 'processCSS'
Finished 'processMarkup'
Finished 'configureEnvironment'
Starting 'buildTypeScript'...
Finished 'buildTypeScript'
Starting 'writeBundles'...
Tracing app...          # where is main, lib, etc?
Tracing aurelia-binding...
Tracing aurelia-bootstrapper...
Tracing aurelia-dependency-injection...
Tracing aurelia-event-aggregator...
Tracing aurelia-framework...
Tracing aurelia-history...
Tracing aurelia-history-browser...
Tracing aurelia-loader-default...
Tracing aurelia-logging-console...
Tracing aurelia-route-recognizer...
Tracing aurelia-router...
Tracing aurelia-templating-binding...
Tracing text...
Tracing aurelia-templating-resources...
Tracing aurelia-templating-router...
Tracing aurelia-testing...
Writing app-bundle.js...
Writing vendor-bundle.js...
Finished 'writeBundles'
```

After:

```bash
$ au build
Starting 'readProjectConfiguration'...
Finished 'readProjectConfiguration'
Starting 'processMarkup'...
Starting 'processCSS'...
Starting 'configureEnvironment'...
Finished 'processCSS'
Finished 'processMarkup'
Finished 'configureEnvironment'
Starting 'buildTypeScript'...
gulp-typescript: ts(tsProject) has been deprecated - use .pipe(tsProject(reporter)) instead
  As of gulp-typescript 3.0, .pipe(ts(tsProject)) should be written as .pipe(tsProject()).
  More information: http://dev.ivogabe.com/gulp-typescript-3/
Finished 'buildTypeScript'
Starting 'writeBundles'...
Tracing app...          # NEW!
Tracing environment...  # NEW!
Tracing main...         # NEW!
Tracing app...
Tracing aurelia-binding...
Tracing aurelia-bootstrapper...
Tracing aurelia-dependency-injection...
Tracing aurelia-event-aggregator...
Tracing aurelia-framework...
Tracing aurelia-history...
Tracing aurelia-history-browser...
Tracing aurelia-loader-default...
Tracing aurelia-logging-console...
Tracing aurelia-pal-browser...
Tracing aurelia-route-recognizer...
Tracing aurelia-router...
Tracing aurelia-templating-binding...
Tracing text...
Tracing aurelia-templating-resources...
Tracing aurelia-templating-router...
Tracing aurelia-testing...
Writing app-bundle.js...
Writing vendor-bundle.js...
Finished 'writeBundles'
```

File: lib/resources/content/tsconfig.json [hunk 1]
```diff
@@ -10,14 +10,21 @@
...
+      ]
+    }
   },
   "exclude": [
     "node_modules"
   ],
   "filesGlob": [
     "./src/**/*.ts",
     "./test/**/*.ts",
+    "./typings/index.d.ts",
```
Comment (JeroenVinke) [inline lib/resources/content/tsconfig.json]: I believe we moved away from typings as of https://github.com/aurelia/cli/pull/457. is this a necessary change or can we do without? Thanks

Comment (sbolel) [inline lib/resources/content/tsconfig.json]: Great catch. Thanks!

**Comment by sbolel** [comment]
https://github.com/aurelia/cli/pull/499

**Pull request closed without merging**